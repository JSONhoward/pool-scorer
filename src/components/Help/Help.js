import React, { useState } from 'react'
import styled from 'styled-components'
import HelpModal from './HelpModal/HelpModal'

const HelpStyled = styled.button`
position: fixed;
bottom: 3.5rem;
left: .5rem;
display: grid;
place-items: center;
height: 3rem;
width: 3rem;
border-radius: 50%;
font-size: 1.5rem;
color: white;
background-color: rgb(0,0,0,.9);
box-shadow: 1px 1px 5px black;
border: none;
cursor: pointer;
`

const Help = ({page}) => {
    const [helpOpen, setHelpOpen] = useState(false)

    const handleModal = () => setHelpOpen(!helpOpen)

    return (
        <>
            <HelpModal handler={handleModal} page={page} open={helpOpen} />
            <HelpStyled data-testid='helpButton' onClick={handleModal}>
                <p>?</p>
            </HelpStyled>
        </>
    )
}

export default Help
