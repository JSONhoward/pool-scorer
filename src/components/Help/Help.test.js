import React from 'react'
import {create} from 'react-test-renderer'
import Help from './Help'

describe('Help Tests', () => {
    test('Snapshot Tests', () => {
        const tree = create(<Help />).toJSON()

        expect(tree).toMatchSnapshot()
    })

    test('Receives Props', () => {
        const component = create(<Help page={'test'} />).root

        expect(component.props).toEqual({page: 'test'})
        
    })
})