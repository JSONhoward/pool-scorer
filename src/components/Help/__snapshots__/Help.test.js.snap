// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Help Tests Snapshot Tests 1`] = `
Array [
  <div
    className="sc-AxhUy cMzBRP"
    open={false}
  >
    <div
      className="sc-AxhCb busWbY"
      data-testid="closeButton"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="2rem"
        size="2rem"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        tabIndex="0"
        viewBox="0 0 352 512"
        width="2rem"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
        />
      </svg>
    </div>
    <p
      className="sc-AxirZ gTVPlk"
    >
      *edit names by clicking on player names
    </p>
    <p
      className="sc-AxjAm iaLvkC"
    >
      Rules
    </p>
  </div>,
  <button
    className="sc-AxgMl eFejOD"
    data-testid="helpButton"
    onClick={[Function]}
  >
    <p>
      ?
    </p>
  </button>,
]
`;
